frontend/js/src/listens/AddToPlaylist.tsx

```diff
@@ -0,0 +1,200 @@
+import * as React from "react";
+import { has } from "lodash";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faFileCirclePlus, faPlusCircle } from "@fortawesome/free-solid-svg-icons";
+import { toast } from "react-toastify";
+import NiceModal, { useModal } from "@ebay/nice-modal-react";
+import { ToastMsg } from "../notifications/Notifications";
+import { PLAYLIST_URI_PREFIX,  listenToJSPFTrack } from "../playlists/utils";
+import GlobalAppContext from "../utils/GlobalAppContext";
+import { getTrackName } from "../utils/utils";
+import CreateOrEditPlaylistModal from "../playlists/CreateOrEditPlaylistModal";
+
+type AddToPlaylistProps = {
+    listen: Listen | JSPFTrack;
+}
+
+export default NiceModal.create((props: AddToPlaylistProps) => {
+    const modal = useModal();
+  
+    const closeModal = React.useCallback(() => {
+      modal.hide();
+      setTimeout(modal.remove, 500);
+    }, [modal]);
+  
+    const { listen } = props;
+    const { APIService, currentUser } = React.useContext(GlobalAppContext);
+    const [playlists, setPlaylists] = React.useState<Array<JSPFObject>>([]);
+    
+    React.useEffect(()=>{
+        async function loadPlaylists() {
+            try {
+                const response = await APIService.getUserPlaylists(
+                    currentUser?.name,
+                    currentUser?.auth_token,
+                    0,
+                    0,
+                    false,
+                    false,
+                );
+                
+                if(!response){
+                    return;
+                }
+                const { playlists: existingPlaylists} = response;
+                setPlaylists(existingPlaylists);
+                window.sessionStorage.setItem("listenbrainz_playlists", JSON.stringify(existingPlaylists));
```

Comment:
Does this session storage cache gets purged somehow? What happens if the user e.g. adds or removes playlists after this was set and then uses the "add to playlist" again? Will they end up with stale cache?

Reply:
Just saw your comment above. Makes some sense, but having to open a new tab still means I will end up with probably stale data after doing changes to playlists and staying in the same tab, right?

Reply:
Yes, I tried it on test.LB and indeed a deleted playlist stays in the cached list and trying to add a track to it results in an error: 
<img width="288" alt="image" src="https://github.com/metabrainz/listenbrainz-server/assets/6179856/a7540fd4-e824-44cb-8117-f6a5f14a3982">

I think this is a case of premature optimization, I'm just going to remove the caching.

For posterity, I would otherwise have to update the stored playlist after modifying a playlist through the website, which involves parsing the stored array, finding and modifying the modified playlist or adding the new one, and storing the stringified result again.
But that still wouldn't solve the (arguably edge) case of deleting a playlist through the API.